creme/neighbors/base_neighbors.py

```diff
@@ -0,0 +1,156 @@
+from collections import deque
+
+import numpy as np
+from sklearn.neighbors import KDTree
+
+
+from creme.utils.skmultiflow_utils import get_dimensions
+
+
+class SlidingWindow:
```

Comment:
There's `utils.Window` in `creme` that is slightly more generic, maybe you can look into using that? Instead of storing two queues you would store pairs of `(x, y)`.

Reply:
@jacobmontiel also suggested me to use `utils.Window` to keep track of `y`. I just did not do it yet because the structure does not currently implement some operations I need (e.g., `popleft`), but I can easily extend it. What do you think @MaxHalford?

We are keeping separate structures for `X` and `y` to avoid creating a `np.ndarray` from a queue every time, though the performed array stacking operations are also expensive. This, indeed, is a temporary solution. In the long run we should explore online learning-friendly approaches, such as Locality Sensitive Hashing.

Reply:
In the latest commit I changed the deque to `utils.Window`.

Reply:
I ended up using a simple `list()` to keep track of `y`, so that I can use the same circular insertion strategy.